sklearn/datasets/openml.py

```diff
@@ -70,22 +99,20 @@ def is_gzip(_fsrc):
 
     local_path = _get_local_path(openml_path, data_home)
     if not os.path.exists(local_path):
-        fsrc = urlopen(req)
         try:
             os.makedirs(os.path.dirname(local_path))
         except OSError:
             # potentially, the directory has been created already
             pass
 
         try:
-            if is_gzip(fsrc):
-                with open(local_path, 'wb') as fdst:
-                    shutil.copyfileobj(fsrc, fdst)
-                    fsrc.close()
-            else:
-                with gzip.GzipFile(local_path, 'wb') as fdst:
-                    shutil.copyfileobj(fsrc, fdst)
-                    fsrc.close()
+            with closing(urlopen(req)) as fsrc:
+                if is_gzip(fsrc):
+                    with open(local_path, 'wb') as fdst:
+                        shutil.copyfileobj(fsrc, fdst)
+                else:
+                    with gzip.GzipFile(local_path, 'wb') as fdst:
+                        shutil.copyfileobj(fsrc, fdst)
         except Exception:
             os.unlink(local_path)
```

Comment:
This line can throw an error if the file in `local_path` does not exists

Reply:
So we should probably only unlink if it exists and otherwise do nothing?

Reply:
This PR was updated to check if the file exists before unlinking it.